dark/reads.py

```diff
@@ -133,6 +135,26 @@ def walkHSP(self, hsp):
             readOffset += 1
             subjectOffset += 1
 
+    def checkAlphabeth(self, count=10):
+        """
+        A function which checks whether the sequence in a L{dark.Read} object
+        corresponds to its readClass. For AA reads, more testing is done in
+        dark.Read.AARead.checkAlphabeth.
+
+        @param count: A C{int} of how many amino acids should be considered.
+        @return: C{True} if the sequence corresponds from its readClass or
+            C{False} if the sequence doesn't correspond.
+        @raise ValueError: If the sequence and the readClass don't match, raise
+            a ValueError.
+        """
+        readLetters = set(self.sequence.upper()[:count])
```

Comment:
On Sun, Apr 12, 2015 at 9:00 PM, Barbara Mühlemann <notifications@github.com
> wrote:

> Do you mean change the default to None?
>
No, just to make it slightly easier to have the whole sequence checked the
None value could be used. It's also possible for the caller just to pass
the length of the sequence as the argument. No big deal either way.